cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -267,24 +259,24 @@ func startControllers(c *cloudcontrollerconfig.CompletedConfig, stopCh <-chan st
 	select {}
 }
 
-// initFunc is used to launch a particular controller.  It may run additional "should I activate checks".
+// InitFunc is used to launch a particular controller.  It may run additional "should I activate checks".
 // Any error returned will cause the controller process to `Fatal`
 // The bool indicates whether the controller was enabled.
-type initFunc func(ctx *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface, stop <-chan struct{}) (debuggingHandler http.Handler, enabled bool, err error)
+type InitFunc func(ctx *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface, stop <-chan struct{}) (debuggingHandler http.Handler, enabled bool, err error)
```

Comment:
I think this (ctx *cloudcontrollerconfig.CompletedConfig) is going to have linking problems when a cloud provider attempts to add a controller which is not in k8s.io/cloud-provider (?new home of cloudcontrollerconfig "k8s.io/kubernetes/cmd/cloud-controller-manager/app/config"?). It would be nice if we our sample added something like NodeIpamController to both see what it takes. It would also demonstrate how a cloud provider should handle config parameters which are not part of k8s.io/cloud-provider.

Also worth looking at https://github.com/kubernetes/kubernetes/blob/release-1.19/cmd/kube-controller-manager/app/controllermanager.go#L357. We seem to be diverging again and it would be nice if we could converge. Might be worth looking at if we can curry in the extra fields we need for the CCM.